File: Behavioural Testing Unit (Android)/Mymou/app/src/main/java/com/example/jbutler/mymou/LogEvent.java
package com.example.jbutler.mymou;

import android.util.Log;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.FileWriter;

/**
 * Writes linked String into  CURRENT_DATE.txt
 */
class LogEvent implements Runnable {

    private final String message;

    public LogEvent(String msg) {
        message = msg;
    }

    @Override
    public void run() {
        //Log.d("LogEvent","Start logging @ " + Thread.currentThread().getName());
        File appFolder = MainMenu.folderManager.getFolder();
        String fileName = MainMenu.folderManager.getBaseDate() + ".txt";
        File saveFile = new File(appFolder, fileName);
        try {
            FileWriter thisWriter = new FileWriter(saveFile, true);
            thisWriter.write("$message\n");
            thisWriter.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
        //Log.d("LogEvent",message + " logged...");
    }

}

File: Behavioural Testing Unit (Android)/Mymou/app/src/main/java/com/example/jbutler/mymou/LogEvent.kt
package com.example.jbutler.mymou

import java.io.File
import java.io.IOException
import java.io.FileWriter

/**
 * Writes linked String into CURRENT_DATE.txt
 */
internal class LogEvent(private val message: String) : Runnable {
    override fun run() {
        val appFolder = MainMenu.folderManager.getFolder()
        val fileName = MainMenu.folderManager.getBaseDate() + ".txt"
        val saveFile = File(appFolder, fileName)
        try {
            FileWriter(saveFile, true).apply {
                write("$message\n\n")
                close()
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}